ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -409,11 +409,13 @@ public void put(MessageWriteSet messageSetToWrite) throws StoreException {
       }
     } catch (StoreException e) {
       throw e;
-    } catch (IOException e) {
-      throw new StoreException("IO error while trying to put blobs to store " + dataDir, e, StoreErrorCodes.IOError);
     } catch (Exception e) {
-      throw new StoreException("Unknown error while trying to put blobs to store " + dataDir, e,
-          StoreErrorCodes.Unknown_Error);
+      if (e.getClass() == IOException.class && e.getMessage().equals(StoreException.IO_ERROR_STR)) {
+        throw new StoreException("IO error while trying to put blobs to store " + dataDir, e, StoreErrorCodes.IOError);
```

Comment:
for e.g., you can just keep a count of the errors (or call into a `StorageManager` provided callback) here.